src/adaptors/adaptor_common.c

```diff
@@ -291,6 +146,29 @@ int qd_raw_connection_write_buffers(pn_raw_connection_t *pn_raw_conn, qd_adaptor
     return num_buffers_written;
 }
 
+int qd_raw_connection_drain_read_write_buffers(pn_raw_connection_t *pn_raw_conn)
+{
+    pn_raw_buffer_t buffs[RAW_BUFFER_BATCH];
+    size_t          n;
+    int             buffers_drained = 0;
+    while ((n = pn_raw_connection_take_written_buffers(pn_raw_conn, buffs, RAW_BUFFER_BATCH))) {
+        for (size_t i = 0; i < n; ++i) {
+            buffers_drained++;
+            qd_adaptor_buffer_t *qd_adaptor_buffer = (qd_adaptor_buffer_t *) buffs[i].context;
+            free_qd_adaptor_buffer_t(qd_adaptor_buffer);
```

Comment:
be consistent when using the api: call qd_adaptor_buffer_free()

Reply:
Done